pkg/migration-tool/utils/client.go

```diff
@@ -160,6 +159,28 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query, desc string) (re
 	return resp.Results[0], len(compressed), len(uncompressed), nil
 }
 
+// PrompbResponse is a type that contains promb-result and information pertaining to it.
+type PrompbResponse struct {
+	ID                   int
+	Result               *prompb.QueryResult
+	NumBytesCompressed   int
+	NumBytesUncompressed int
+}
+
+// ReadChannels calls the Read and responds on the channels.
+func (c *Client) ReadChannels(ctx context.Context, query *prompb.Query, shardID int, desc string, doneChan chan<- *PrompbResponse, errChan chan<- error) {
+	result, numBytesCompressed, numBytesUncompressed, err := c.Read(ctx, query, desc)
+	if err != nil {
+		errChan <- fmt.Errorf("read-channels: %w", err)
```

Comment:
important: I believe the user of this call expects either an error response or a done response but not both
```suggestion
		errChan <- fmt.Errorf("read-channels: %w", err)
		return
```

Reply:
Thank you for pointing this out.